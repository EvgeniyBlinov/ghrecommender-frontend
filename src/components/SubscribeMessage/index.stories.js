import React from 'react'
import { storiesOf, action } from '@kadira/storybook'
import 'semantic-ui-css/semantic.min.css'

import SubscribeMessage from './index'

const props = {
  onClick: action('clicked'),
}

const stories = storiesOf('Component - <SubscribeMessage />', module)

stories.add('<SubscribeMessage />', () => <SubscribeMessage {...props} />)

stories.add('<SubscribeMessage subscribed />', () => (
  <SubscribeMessage {...props} subscribed={true} />
))

stories.add('<SubscribeMessage loading />', () => <SubscribeMessage {...props} loading={true} />)

stories.add('<SubscribeMessage loading subscribed />', () => (
  <SubscribeMessage {...props} loading={true} subscribed={true} />
))
