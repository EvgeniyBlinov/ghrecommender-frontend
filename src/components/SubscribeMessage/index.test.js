import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SubscribeMessage from './index'

const props = {
  onClick: jest.fn(),
}

describe('<SubscribeMessage />', () => {
  test('<SubscribeMessage />', () => {
    const component = renderer.create(<SubscribeMessage {...props} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('<SubscribeMessage loading />', () => {
    const component = renderer.create(<SubscribeMessage {...props} loading />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('<SubscribeMessage subscribed />', () => {
    const component = renderer.create(<SubscribeMessage {...props} subscribed />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('<SubscribeMessage subscribed loading />', () => {
    const component = renderer.create(<SubscribeMessage {...props} subscribed loading />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('onClick', () => {
    const component = shallow(<SubscribeMessage {...props} subscribed loading />)
    component.find('Button').simulate('click')
    expect(props.onClick.mock.calls.length).toBe(1)
  })
})
