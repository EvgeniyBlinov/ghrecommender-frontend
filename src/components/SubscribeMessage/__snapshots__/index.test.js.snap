// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubscribeMessage /> <SubscribeMessage /> 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="mail outline icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small positive button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Subscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessage /> <SubscribeMessage loading /> 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="circle notched loading icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small loading positive button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Subscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessage /> <SubscribeMessage subscribed /> 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="mail outline icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small negative button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Unsubscribe
    </button>
  </div>
</div>
`;

exports[`<SubscribeMessage /> <SubscribeMessage subscribed loading /> 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="circle notched loading icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Subscribe to get new recommendations as soon as they arrive.
    </div>
    <br />
    <button
      className="ui small loading negative button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Unsubscribe
    </button>
  </div>
</div>
`;
